test/fake-intake/fakeintake/server.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fakeintake
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+type payload struct {
+	timestamp time.Time
+	data      []byte
+}
+
+type FakeIntake struct {
+	mu     sync.RWMutex
+	server http.Server
+
+	payloadStore map[string][]payload
+}
+
+// NewFakeIntake creates a new fake intake and starts it on localhost:5000
+func NewFakeIntake() *FakeIntake {
+	fi := &FakeIntake{
+		mu:           sync.RWMutex{},
+		payloadStore: map[string][]payload{},
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", fi.postPayload)
+	mux.HandleFunc("/fake/payloads/", fi.getPayloads)
+
+	fi.server = http.Server{
+		Addr:    ":5000",
+		Handler: mux,
+	}
+
+	fi.start()
+
+	return fi
+}
+
+func (fi *FakeIntake) start() {
+	go func() {
+		err := fi.server.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			fmt.Printf("Error creating fake intake server at %s: %v", fi.server.Addr, err)
+		}
+	}()
+}
+
+// Stop Gracefully stop the http server
+func (fi *FakeIntake) Stop() error {
+	return fi.server.Shutdown(context.Background())
+}
+
+type postPayloadResponse struct {
+	Errors []string `json:"errors"`
+}
+
+func (fi *FakeIntake) postPayload(w http.ResponseWriter, req *http.Request) {
+	fmt.Println("Handling PostPayload request")
+	if req == nil {
+		writePostPayloadResponse(w, []string{"invalid request, nil request"})
+		return
+	}
+
+	if req.Method != http.MethodPost {
+		writePostPayloadResponse(w, []string{fmt.Sprintf("invalid request with route %s and method %s", req.URL.Path, req.Method)})
```

Comment:
nitpick: It would be great to return an error code for all kind of errors. For example [405](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) in this case.

Reply:
Changing this, will allow `GET` methods too to accept any get request